cmd/package-operator-manager/main.go

```diff
@@ -375,24 +378,57 @@ func runManager(log logr.Logger, scheme *runtime.Scheme, opts opts) error {
 		return fmt.Errorf("unable to create controller for ClusterPackage: %w", err)
 	}
 
-	// Probe for HyperShift API
-	hostedClusterGVK := hypershiftv1beta1.GroupVersion.WithKind("HostedCluster")
-	_, err = mgr.GetRESTMapper().RESTMapping(hostedClusterGVK.GroupKind(), hostedClusterGVK.Version)
-	if !meta.IsNoMatchError(err) {
-		// HyperShift HostedCluster API is present on the cluster
-		// Auto-Enable HyperShift integration controller:
-		log.Info("detected HostedCluster API, enabling HyperShift integration")
-		if err = hostedclusters.NewHostedClusterController(
-			mgr.GetClient(), ctrl.Log.WithName("controllers").WithName("HostedCluster"), mgr.GetScheme(),
-			opts.remotePhasePackageImage,
-		).SetupWithManager(mgr); err != nil {
-			return fmt.Errorf("unable to create controller for HostedCluster: %w", err)
+	go func() {
+		err := handleHyperShift(ctx, log, mgr, opts)
+		switch {
+		case err == nil:
+		case ctx.Err() != nil && errors.Is(err, ctx.Err()):
+		default:
+			// Don't judge, nico says its ok.
+			os.Exit(1)
```

Comment:
This is not needed if booting up the controller works during runtime.
We only need to exit if the controller MUST be started during app boostrap